lib/modules/cardiovascular_disease.rb

```diff
@@ -356,7 +356,48 @@ class CardiovascularDisease < Synthea::Rules
           end
         end
       end
+      #-----------------------------------------------------------------------#
+      #Treatments and Medications
+      #-----------------------------------------------------------------------#
+      rule :heart_healthy_lifestyle, [:coronary_heart_disease, :stroke, :cardiac_arrest, :myocardial_infarction], [] do |time, entity|
+        reasons = []
+        [:coronary_heart_disease, :stroke, :cardiac_arrest, :myocardial_infarction].each do |disease|
+          reasons << disease if entity[disease] || entity.had_event?(disease)
+        end
+        puts entity[:name_last] if reasons.length > 1
+        unless reasons.empty?
+          entity[:careplan] ||= Hash.new
+          entity[:careplan][:cardiovascular_disease] ||= {'activities'=>[:exercise, :stress_management, :stop_smoking, :healthy_diet]}
+          entity[:careplan][:cardiovascular_disease]['reasons'] = reasons
+        end
+      end
 
+      rule :chd_treatment, [:coronary_heart_disease], [:coronary_heart_disease] do |time, entity|
+        meds = [:clopidogrel, :simvastatin, :amlodipine, :nitroglycerin]
+        if entity[:coronary_heart_disease]
+          entity[:medications] ||= Hash.new
+          meds.each {|m| prescribeMedication(m, :coronary_heart_disease, time, entity)}
+        elsif entity[:medications]
+          meds.each do |m|
+            stopMedication(m, :coronary_heart_disease, time, entity) if entity[:medications][m]
+          end
+        end
+      end
+
+      rule :atrial_fibrillation_treatment, [:atrial_fibrillation], [:atrial_fibrillation] do |time, entity|
+        meds = [:warfarin, :verapamil, :digoxin]
+        if entity[:atrial_fibrillation]
+          entity[:medications] ||= Hash.new
+          meds.each {|m| prescribeMedication(m, :atrial_fibrillation, time, entity)}
+
+          entity[:cardiovascular_procedures] ||= Hash.new
+          entity[:cardiovascular_procedures][:atrial_fibrillation] ||= [:electrical_cardioversion, :catheter_ablation]
```

Comment:
I'm not sure that *every* afib patient is going to get all of those meds *plus* electrical cardioversion, *plus* catheter ablation.  This is probably not realistic.  According to WebMD, electrical cardioversion is only done right away if afib is discovered in its first 48 hours -- otherwise it might be weeks after the meds.  Catheter ablation seems even more rare, perhaps only being done when medication is proving ineffective or when the patient refuses the medication.  We may want to space some of these out by time and/or only apply them to selective people (not all afib patients).